### PR TITLE
New Data Source: `azurerm_route_table`

### LINKED ISSUE #1155: New Data Source: `azurerm_route_table`
Hi there,
can a data source for "azurerm_route_table" can be added?
We use a script acs-engine to deploy kubernetes cluster (so without terraform) and this autocreate a route table, problem is: all infrastructure vnet/subnet are already created by terraform, so without data source added, it will remove the route table if we tf apply.
Today we add the route_table_id in a static variable in the subnet creation file, but since they happen to destroy and recreate cluster, this is not dynamic...
Thank you

### PR DESCRIPTION
**Pull request by tombuildsstuff**
Tests pass:

```
$ acctests azurerm TestAccDataSourceAzureRMRouteTable_
=== RUN   TestAccDataSourceAzureRMRouteTable_basic
--- PASS: TestAccDataSourceAzureRMRouteTable_basic (87.04s)
=== RUN   TestAccDataSourceAzureRMRouteTable_singleRoute
--- PASS: TestAccDataSourceAzureRMRouteTable_singleRoute (67.25s)
=== RUN   TestAccDataSourceAzureRMRouteTable_multipleRoutes
--- PASS: TestAccDataSourceAzureRMRouteTable_multipleRoutes (77.71s)
PASS
ok  	github.com/terraform-providers/terraform-provider-azurerm/azurerm	232.032s
```

Fixes #1155

### CODE REVIEW

File: azurerm/data_source_route_table.go [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+		Read: dataSourceArmRouteTableRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"resource_group_name": resourceGroupNameSchema(),
```
Comment (katbyte) [inline azurerm/data_source_route_table.go]: Should this be `resourceGroupNameForDataSourceSchema()`?

File: website/docs/d/route_table.html.markdown [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+* `name` - (Required) The name of the Route Table
+
+* `resource_group_name` - (Required) The name of the Resource Group in which the Route Table exists
+
+## Attributes Reference
+
+The following attributes are exported:
+
+* `id` - The Route Table ID.
```
Comment (katbyte) [inline website/docs/d/route_table.html.markdown]: Some lines end with a `.`, some do not

Comment (tombuildsstuff) [reply website/docs/d/route_table.html.markdown]: 👍 fixed

File: azurerm/data_source_route_table.go (hunk 1 above)
Comment (tombuildsstuff) [reply azurerm/data_source_route_table.go]: 👍 fixed

**Pull request merged**